requirements/nightly.in

```diff
@@ -4,7 +4,7 @@
 dask
 graphviz
 platformdirs<4
-https://github.com/scipp/scipp/releases/download/nightly/scipp-nightly-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
+https://github.com/scipp/scipp/releases/download/nightly/scipp-nightly-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
```

Comment:
For some reason, the version is still 3.9 in the `nightly.txt` file. Is this not updated when we run `tox -e deps`?

Reply:
Ah yes nice catch, the issue here is that I'm using a Mac to run the tox -e deps and it would throw an error while trying to add this to nightly.txt (`scipp-nightly-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl is not a supported wheel on this platform.`). So we need someone (or the CI) with x86 linux to run this action.

Reply:
> So we need someone (or the CI) with x86 linux to run this action.

Ah, I never thought of this! Discrimination against Mac folk... I think requiring someone with linux to do it is not a viable solution in the long term. Using an action is annoying because it's something you want to do locally.
What else could we do?

Reply:
Apparently it is possible to have conditional requirements in the `requirements.txt` files.
https://pip.pypa.io/en/stable/reference/requirement-specifiers/#requirement-specifiers
Should we use this syntax?

Reply:
Using the syntax will resolve the local errors on a non linux machine but it won't update the nightly.txt file as a mac woudn't know how to resolve a linux wheel :/

Reply:
You mean the `pip-compile` doesn't know how to write the conditional requirements from `.in` files...?

Reply:
Also we are not building scipp in nightly for Mac OS or Windows. Maybe we should.

Reply:
> You mean the pip-compile doesn't know how to write the conditional requirements from .in files...?

It needs to resolve on my local machine too, and (I think!) it would just ignore a platform specific requirements.

>  Also we are not building scipp in nightly for Mac OS or Windows. Maybe we should.

I think we should if it's not too much work!